src/pages/stop-details/stop-details.ts

```diff
@@ -15,12 +18,23 @@ import { IonicPage, NavParams, ViewController } from 'ionic-angular';
 })
 export class StopDetailsPage {
   public stopName: string;
-
-  constructor(public navParams: NavParams, public view: ViewController) {
+  public stopLine: string;
+  public stopID: string;
+  stopTime: Observable<any[]>;
+  constructor(public navParams: NavParams, public view: ViewController, public firebaseProvider: FirebaseProvider) {
   }
 
   ionViewDidLoad() {
     this.stopName = this.navParams.data.stopName;
+    this.stopLine = this.navParams.data.stopLine;
+    var d = new Date();
+    this.stopID = this.navParams.data.stopID;
+    this.stopTime = this.firebaseProvider.getstopTimes(this.stopID, d.getDay()).valueChanges();
```

Comment:
Since this is the stop details view for an individual stop, why don't we get all details of a stop from firebase here. Then, for example on the client we can just take the location from the blob when we need location, we can take times when we need times, etc.